main.py

```diff
@@ -66,10 +67,13 @@ def log_starting_thread(thread: Thread):
 
 def get_infinite_iterator(collection: list):
     length = len(collection)
-    i = 0
+    i = (cfg["starting_core"] - 1) * 2
```

Comment:
This is a nice calculation and works well for the current Ryzen processors! I see you are using a 1-indexed core number, so I would suggest adding that to the comment at the top.
Also, how about we change this calculation depending on if hyperthreading is enabled?
```suggestion
    i = (cfg["starting_core"] - 1) * (cfg["hyper_threading"] + 1)
```
Here I use the fact that Python implicitly casts booleans to integers if you put them in 'calculations'. False becomes 0, True becomes 1. Thus, if hyper threading is set to True, this last factor evaluates to 2, otherwise it is 1.